File: Homework/JavaScript-Fundamentals/11. Strings/correct-brackets.js
function solve(args) {

    console.log(checkBrackets(args[0]));

    function checkBrackets(str) {
        var stack = [];
        for (let i = 0, len = str.length; i < len; i += 1) {
            if (str[i] === ')') {
                if (stack.length === 0) {
                    return 'Incorrect';
                } else {
                    stack.pop();
                }
            }

            if (str[i] === '(') {
                stack.push('(');
            }
        }

        if (stack.length !== 0) {
            return 'Incorrect';
        } else {
            return 'Correct';
        }
    }
}

File: Homework/CSharp-Part-2/06. Strings-and-Text-Processing/CorrectBrackets.cs
using System;
using System.Collections.Generic;

class CorrectBrackets
{
    static void Main()
    {
        string input = Console.ReadLine();
        char opening = '(';
        char closing = ')';
        Stack<char> brackets = new Stack<char>();
        bool areBracketsCorrect = true;

        for (int i = 0; i < input.Length; i++)
        {
            char currentSymbol = input[i];
            if (currentSymbol == opening)
            {
                brackets.Push(currentSymbol);
            }

            if (currentSymbol == closing)
            {
                if (brackets.Count == 0)
                {
                    areBracketsCorrect = false;
                    break;
                }
                else
                {
                    brackets.Pop();
                }
            }
        }

        if (brackets.Count > 0)
        {
            areBracketsCorrect = false;
        }

        Console.WriteLine(areBracketsCorrect ? "Correct" : "Incorrect");
    }
}